src/bot/commands/addgreeting.ts

```diff
@@ -0,0 +1,55 @@
+import Discord, { Collection } from 'discord.js';
+
+import MongoService from '../../core/services/mongo.service';
+import ConfigService from '../../core/services/config.service'
+
+import container from '../../inversity.config';
+
+import ICommand from './ICommand';
+
+const prefix = container.resolve<ConfigService>(ConfigService).get('BOT_TRIGGER');
+
+const command: ICommand = {
+  name: 'addgreeting',
+  aliases: ['ag'],
+  description: 'Adds a string to the list greetings used when new users connect to server! Include `{name}` in your message to replace with the new users name.',
+  example: `\`${prefix}addgreeting Welcome to the club {name}\``,
+  async execute(message: Discord.Message, args: string[], prefix?: string, commands?: Collection<string, ICommand>, dbService?: MongoService) {
+    let isPermitted = false;
+    const permittedRoles = ['bot-devs', 'admin', 'staff']
+
+    message.guild.roles.cache.forEach(r => {
+      if (permittedRoles.indexOf(r.name) !== -1) {
+        isPermitted = true;
+      }
+    })
```

Comment:
Two points:
1. This can be shortened and short-circuited, by using `Array.prototype.some`
1. I believe in Discord.js ([docs](https://discord.js.org/#/docs/main/stable/class/Message?scrollTo=guild)) the `message.guild.roles` object doesn't give you the guilds of the user, but the roles of the guild. If you wanna check permissions, we should use `message.member.roles.cache` [[cit1]](https://stackoverflow.com/questions/56926998/how-can-i-check-if-the-message-author-has-an-admin-role-using-discord-js) [[cit2]](https://discord.js.org/#/docs/main/stable/class/GuildMember?scrollTo=roles)

In summary, this should be what we are looking for:
```suggestion
    const isPermitted = message.member.roles.cache.some(r => permittedRoles.indexOf(r.name) !== -1);
```

This also prevents the unnecessary runs over the permissions after the flag is found.
Do we want to move this behavior into a utility? We will probably be using it a lot.

Check my suggestion code works, I haven't tested it. But it should.